pkg/instrumentation/sdk.go

```diff
@@ -263,8 +263,22 @@ func (i *sdkInjector) injectCommonSDKConfig(ctx context.Context, otelinst v1alph
 		}
 	}
 
+	// get existing resources env var and parse it into a map
+	existingRes := map[string]string{}
+	existingResourceEnvIdx := getIndexOfEnv(pod.Spec.Containers[appIndex].Env, constants.EnvOTELResourceAttrs)
```

Comment:
Looks like `createResourceMap` was already doing this and reading the env var to put it in `resourceMap`. So why wasnt that check i.e. `resourceMap[key] == ""` not sufficient for things like K8SPodNameKey?

Reply:
for k, v := range otelinst.Spec.Resource.Attributes {
		if !existingRes[k] {
			res[k] = v
		}
	}
If the attribute doesn't exist in existingRes, then the res will be updated. So when the second injection happens, res doesn't have the semconv.K8SPodNameKey key in it as it was existing value. So just checking the resourceMap[string(semconv.K8SPodNameKey)] == "" will be true and the key is added again